queue-non-urgent-requests.md

```diff
@@ -0,0 +1,43 @@
+---
+version: 1.0
+submitted_by: greenhsu123
+published_date: TBD
+category: cloud
+tags: 
+ - role:cloud-engineer
+ - size:small
+---
+
+# Queuing non-urgent processing requests
+
+## Description
+
+All systems have periods of peak and low load. From a hardware-efficiency perspective, we are more efficient with hardware if we minimise the impact of requests spikes with an implementation that allows an even utilisation of components. From an energy-efficiency perspective, we are more efficient with energy if we ensure that idle resources are kept to a minimum. 
+
+## Solution
+
+Utilise a message queue, so requests that do not require immediate processing are queued. 
+
+## SCI Impact
+
+`SCI = (E * I) + M per R`  
+[Sofware Carbon Intensity Spec](https://github.com/Green-Software-Foundation/software_carbon_intensity)
+
+Concerning the SCI equation, optimising peak CPU utilisation will impact two parts:
+
+- `E`: By reducing the total number of computing equipment required, we reduce the total embodied carbon.
+- `M`: By reducing the total number of idle resources, we reduce the total electricity required.
+
+## Assumptions
+- There is an assumption that the components in your system can communicate with a message queue asynchronously. If your component is the producer, it can add requests to the queue without waiting for the previous sent to be processed. If your component is the consumer, it will only process requests when they are available. We assume that no components in the system ever stalled while waiting for another. 
+
+
+## Considerations
+- Decoupling components of your system can introduce unnecessary complexity as well as managing a message queue for your system can introduce unnecessary overhead, so careful consideration should be taken to evaluate if a message queue is suited for your system 
+
+
+## References
```

Comment:
There is a similar pattern in pull request, should we link them in their respective Reference sections?

https://github.com/Green-Software-Foundation/green-software-patterns/pull/86

Reply:
I think we should, but should we link a PR though...

Reply:
agreed